### PR TITLE
Update dune-project with dependency constraints

### PR DESCRIPTION
**Pull request by mjambon**
The `dune-project` file is used by dune to produce the opam files. These changes bring the opam files closer to what's needed by opam-repository. See https://github.com/ocaml/opam-repository/pull/20905 to get a sense of what happened.

The next release, which I'm going to make very soon (2.3.3), is going to be much easier.

### PR checklist

- [x] New code has tests to catch future regressions
- [x] Documentation is up-to-date
- [x] `CHANGES.md` is up-to-date

### CODE REVIEW

File: atd.opam [hunk 1]
```diff
@@ -32,8 +32,8 @@ homepage: "https://github.com/ahrefs/atd"
 bug-reports: "https://github.com/ahrefs/atd/issues"
 depends: [
   "ocaml" {>= "4.08"}
-  "dune" {>= "2.0"}
-  "menhir"
+  "dune" {>= "2.8" & >= "2.8"}
```
Comment (Khady) [inline atd.opam:35]: don't need the double constraint here

File: atdgen-runtime.opam [hunk 2]
```diff
@@ -30,7 +30,7 @@ homepage: "https://github.com/ahrefs/atd"
 bug-reports: "https://github.com/ahrefs/atd/issues"
 depends: [
   "ocaml" {>= "4.08"}
-  "dune" {>= "2.0"}
+  "dune" {>= "2.8" & >= "2.8"}
```
Comment (Khady) [inline atdgen-runtime.opam:33]: same

File: dune-project [hunk 3]
```diff
@@ -35,8 +35,8 @@
  (name atd)
  (depends
   (ocaml (>= 4.08))
-  (dune (>= 2.0))
-  menhir
+  (dune (>= 2.8))
```
Comment (Khady) [inline dune-project:38]: you can remove all the `(dune (>= 2.8))` from dune-project because dune automatically adds this dependency

Comment (mjambon) [reply dune-project:38]: ok, great. The fix is here: https://github.com/ahrefs/atd/pull/260

**Pull request merged**